### PR TITLE
feat: add extraContainers argument to deployment

### PR DESCRIPTION
**Pull request by simonfr**
Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by simonfr** [comment]
@spencergilbert @seanson Can you review my PR please ? :-)

File: charts/argo-cd/templates/argocd-server/deployment.yaml [hunk 1]
```diff
@@ -120,6 +120,9 @@ spec:
           failureThreshold: {{ .Values.server.readinessProbe.failureThreshold }}
         resources:
 {{- toYaml .Values.server.resources | nindent 10 }}
+      {{- if .Values.server.extraContainers }}
```
Comment (seanson) [inline charts/argo-cd/templates/argocd-server/deployment.yaml:123]: Could we align the Helm template as per the indent above.

File: charts/argo-cd/values.yaml [hunk 2]
```diff
@@ -653,6 +653,8 @@ server:
   #      oauthclientCredentials:
   #        secretName: argocd-secret
 
+  extraContainers: []
```
Comment (seanson) [inline charts/argo-cd/values.yaml:656]: Could we get an example of an extra container in the comments below this entry here so the purpose is clear.

**Review by seanson — CHANGES_REQUESTED** [review]
Hi @simonfr, thanks for the PR! A few requests and it should be good to go.

File: charts/argo-cd/templates/argocd-server/deployment.yaml (hunk 1 above)
Comment (simonfr) [reply charts/argo-cd/templates/argocd-server/deployment.yaml]: It's done :-)

File: charts/argo-cd/values.yaml (hunk 2 above)
Comment (simonfr) [reply charts/argo-cd/values.yaml:656]: Added :-)

**Review by seanson — APPROVED** [review]
Awesome, thanks @simonfr!

**Pull request merged**